### PR TITLE
Th refactor scripts

### LINKED ISSUE #14: Review/Refactor scripts file
* Review and refactor the scripts.js file

### PR DESCRIPTION
**Pull request by relyt4me**
### What’s this PR do?
Going line by line moved all of the global documentbyID, Created a single function for the List HTML maker, renamed functions for better readability, Create global variables for only the repositories and variable used through multiple functions.
### Where should the reviewer start?
All changes made in scripts js follow from the startApp functions

### How should this be manually tested?
page runs on npm start

### Any background context you want to provide?
Worked with this on call for clearer understanding between classes
comments in code represent changes and remove any that have been requested

### What are the relevant tickets?
resolves #14

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -15,110 +15,80 @@ import Hydration from './Hydration';
...
-var userStairsThisWeek = document.getElementById('userStairsThisWeek');
-var userMinutesThisWeek = document.getElementById('userMinutesThisWeek');
-var bestUserSteps = document.getElementById('bestUserSteps');
-var streakList = document.getElementById('streakList');
-var streakListMinutes = document.getElementById('streakListMinutes');
+let userRepo;
+let hydrationRepo;
+let sleepRepo;
+let activityRepo;
+let currentUser;
```
Comment (jordy1611) [inline src/scripts.js:22]: You're my hero. Great job eliminating global variables

File: src/scripts.js [hunk 2]
```diff
@@ -15,110 +15,80 @@ import Hydration from './Hydration';
...
-
-function makeUsers(array) {
-  //rename instatiateUsers remove parameter add line under with returned array
-  userData.forEach(function (dataItem) {
-    //this is a data file and makes instances of all of the users from the data file in one array
-    let user = new User(dataItem); // need to get data file from the API
-    array.push(user);
-  });
+  instantiatePageData();
+  populatePage();
```
Comment (jordy1611) [inline src/scripts.js:28]: This is significantly cleaner

File: src/scripts.js [hunk 3]
```diff
@@ -127,99 +97,53 @@ function makeWinnerID(activityInfo, user, dateString, userStorage) {
...
-  streakListMinutes.insertAdjacentHTML('afterBegin', makeStepStreakHTML(id, activityInfo, userStorage, activityInfo.getStreak(userStorage, id, 'minutesActive')));
-  friendChallengeListHistory.insertAdjacentHTML('afterBegin', makeFriendChallengeHTML(id, activityInfo, userStorage, activityInfo.showChallengeListAndWinner(user, dateString, userStorage)));
-  bigWinner.insertAdjacentHTML('afterBegin', `THIS WEEK'S WINNER! ${activityInfo.showcaseWinner(user, dateString, userStorage)} steps`);
+function populateFriendsSection(currentDate) {
+  document.getElementById('friendChallengeListToday').insertAdjacentHTML('afterBegin', makeFriendChallengeHTML(activityRepo.showChallengeListAndWinner(currentUser, currentDate, userRepo)));
+  document.getElementById('streakList').insertAdjacentHTML('afterBegin', makeStepStreakHTML(activityRepo.getStreak(userRepo, currentUser.id, 'numSteps')));
+  document.getElementById('streakListMinutes').insertAdjacentHTML('afterBegin', makeStepStreakHTML(activityRepo.getStreak(userRepo, currentUser.id, 'minutesActive')));
```
Comment (jordy1611) [inline src/scripts.js:139]: These have nothing to do with friends and should probably be instantiated in the populateActivity Section

File: src/scripts.js [hunk 4]
```diff
@@ -127,99 +97,53 @@ function makeWinnerID(activityInfo, user, dateString, userStorage) {
...
-  return method.map((friendChallengeData) => `<li class="historical-list-listItem">Your friend ${friendChallengeData} average steps.</li>`).join('');
+function makeFriendChallengeHTML(arrayData) {
+  return arrayData.map((friendChallengeData) => `<li class="historical-list-listItem">Your friend ${friendChallengeData} average steps.</li>`).join('');
 }
 
-function makeStepStreakHTML(id, activityInfo, userStorage, method) {
-  return method.map((streakData) => `<li class="historical-list-listItem">${streakData}!</li>`).join('');
+function makeStepStreakHTML(arrayData) {
+  return arrayData.map((streakData) => `<li class="historical-list-listItem">${streakData}!</li>`).join('');
 }
```
Comment (jordy1611) [inline src/scripts.js:149]: This should probably be tied to populateActivity

File: src/scripts.js (hunk 1 above)
Comment (carlymclift) [reply src/scripts.js:22]: This will make everything look so much better and more readble.

File: src/scripts.js (hunk 3 above)
Comment (relyt4me) [reply src/scripts.js:139]: Good point ill add a comment

File: src/scripts.js [hunk 5]
```diff
@@ -127,99 +97,53 @@ function makeWinnerID(activityInfo, user, dateString, userStorage) {
...
+  document.getElementById('hydrationAverage').insertAdjacentHTML('afterBegin', `<p>Your average water intake is</p><p><span class="number">${hydrationRepo.calculateAverageOunces(currentUser.id)}</span></p> <p>oz per day.</p>`);
+  document.getElementById('hydrationThisWeek').insertAdjacentHTML('afterBegin', makeArrayIntoHTMLList(hydrationRepo.calculateFirstWeekOunces(userRepo, currentUser.id), 'oz'));
 }
 
-// All of these are the same method for HTML listing if we send in units (variable should not be method)
-function makeStepsHTML(id, activityInfo, userStorage, method) {
-  return method.map((activityData) => `<li class="historical-list-listItem">On ${activityData} steps</li>`).join('');
+// ['2019/4/20 32', '2019/4/20 32', '2019/4/20 32', '2019/4/20 32'] - array format for these functions
+function makeArrayIntoHTMLList(arrayData, unit) {
+  return arrayData.map((dateAndAmount) => `<li class="historical-list-listItem">On ${dateAndAmount}${unit}</li>`).join('');
```
Comment (carlymclift) [inline src/scripts.js:109]: This function is going to be a life saver. Really clean and DRY code here

File: src/scripts.js (hunk 4 above)
Comment (relyt4me) [reply src/scripts.js:149]: Thank you, I will add a comment for the next refactor

File: src/scripts.js (hunk 5 above)
Comment (relyt4me) [reply src/scripts.js:109]: Still thinking of a better name for it or for the parameters if you happen to think of one later.

**Pull request merged**